frontend/src/metabase/query_builder/components/template_tags/TagEditorParam.jsx

```diff
@@ -337,7 +339,10 @@ export class TagEditorParam extends Component {
                     }
               }
               value={tag.default}
-              setValue={value => this.setParameterAttribute("default", value)}
+              setValue={value => {
+                this.setParameterAttribute("default", value);
+                this.props.setParameterValue(tag.id, value);
```

Comment:
This changes the parameter value when the user changes its default value to match it. This is by design, as presumably the user wants to run the query with the newly set default value.